python/cudf/cudf/dataframe/index.py

```diff
@@ -67,17 +67,17 @@ def take(self, indices):
         ---
         indices: An array-like that maps to values contained in this Index.
         """
-        assert indices.dtype.kind in "iu"
-        if indices.size == 0:
-            # Empty indices
-            return RangeIndex(indices.size)
-        else:
-            # Gather
-            index = cpp_copying.apply_gather_array(self.gpu_values, indices)
-            col = self.as_column().replace(data=index.data)
-            new_index = as_index(col)
-            new_index.name = self.name
-            return new_index
+        from cudf.dataframe.series import Series
+
+        # Gather
+        indices = Series(indices)
```

Comment:
We don't need a Series here, only a column.
```suggestion
        indices = columnops.as_column(indices)
```

Reply:
thanks